erpnext/controllers/accounts_controller.py

```diff
@@ -237,8 +237,8 @@ def set_missing_item_details(self, for_validate=False):
 								item.set(fieldname, value)
 
 							elif fieldname == "serial_no":
-								stock_qty = item.get("stock_qty") * -1 if item.get("stock_qty") < 0 else item.get("stock_qty")
-								if stock_qty != len(get_serial_nos(item.get('serial_no'))):
+								item_qty = abs(item.get("qty"))
```

Comment:
Serial Nos are maintained based on Stock UOM, hence it will not work if sales UOM is different than stock UOM. May be you can use `item_qty = abs(item.get("qty")) * item.get("conversion_factor")`